CIP-????/README.md

```diff
@@ -0,0 +1,334 @@
+We use this to enable “unresolved transaction inputs”, and also atomic groups of transactions. 
+We show how these features can be used to address a number of use cases from CPS-????.
+
+## Motivation: why is this CIP necessary?
+
+This CIP provides a partial solution to the problems described in CPS-????. 
+In particular, it describes some ledger changes that allow intent settlement for a wide range of intents that require “counterparty irrelevance”, including most of the swap use cases and dApp fee sponsorship.
+
+## Specification
+
+### Validation zones
+
+A validation zone is a container for transactions. 
+The main property of a validation zone is that the node may need to look at all the transactions in the zone in order to decide if any of them are valid. 
+Within validation zones we can therefore allow features that require “earlier” transactions to depend on “later” transactions for their validity.
+
+At the ledger level, a validation zone:
+
+- Is a new top-level element (an alternative to a transaction)
+- Contains a sequence of transactions (not zones, there is no point nesting them)
+- Carries no further data beyond this
+- Triggers a special mode of validation for the transactions within (see “validating a zone” below)
+
+### Unresolved inputs and resolving outputs
+
+We add a new kind of transaction input, an “unresolved input” (UTXI). 
+An unresolved input differs from a normal transaction input:
+
+- It does not specify a transaction output that it spends
+- It does specify the value that the input provides
+- It has a credential and an optional datum
+
+Unresolved inputs must be addressable. 
+Current transaction inputs are not addressable by the combination of transaction ID and an index, since they are kept in a set. 
+We therefore need to implement https://github.com/cardano-foundation/CIPs/pull/758, or to keep the unresolved inputs in a separate list.
+
+We add a new kind of transaction output, a “resolving output”. 
+A resolving output differs from a normal transaction output:
+- It specifies that it “resolves” a particular unresolved transaction input, by specifying the transaction ID and index of the input
+- It does not specify the value that it provides
+- It does not have a credential
+
+Unresolved inputs and resolving outputs are illegal outside a validation zone. 
+They are included in the script context accurately, with one subtlety: in the same way that normal transaction inputs are decorated with the value that they provide, resolving transaction outputs are similarly decorated with the value that they provide.
+
+### Required transactions
+
+We add a new field to the transaction body, `requiredTxs`, which is a list of transaction IDs.
+
+The `requiredTxs` field is special in that it is _not_ included when hashing the transaction body to create the transaction ID. 
+It is however signed when the transaction body is signed.
+
+The required transactions field is illegal outside a validation zone. 
```

Comment:
I still think that we should never be outside validation zone, single transactions are just a zone of size 1

Reply:
I guess we could do that. It makes the top-level structure more regular. But also the only stuff it allows is pointless:
- Transactions requiring themselves, which is legal but pointless
- Transactions fulfilling their own inputs, which is illegal

Reply:
i don't get how the second one is allowed. Also, I definitely think making a more regular structure is good here.

Reply:
So long as this is a logical abstraction, and doesn't add overhead on the wire for every tx